bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java

```diff
@@ -0,0 +1,220 @@
+     * @return Stub server listen port or {@code -1} if N/A
+     */
+    public int getStubServerPort() {
+        return this.stubServerPort;
+    }
+
+    /**
+     * Get the stub server listen IP addresses (from hostnames)
+     *
+     * @return A set of listen addresses
+     * @throws ArgoConfigurationException if at least one of the {@code stubServerListenAddresses} is a hostname and
+     *             cannot be resolved to an IP address
+     */
+    public Set<InetAddress> getStubServerListenAddresses() throws ArgoConfigurationException {
+        var addresses = new LinkedHashSet<InetAddress>();
+        for (var t : stubServerListenAddresses) {
+            try {
+                addresses.add(Objects.requireNonNull(InetAddress.getByName(t)));
+            } catch (UnknownHostException e) {
+                throw ArgoConfigurationException.forInvalidParamValue(
+                        ArgoClimaBindingConstants.PARAMETER_STUB_SERVER_LISTEN_ADDRESSES, t, i18nProvider, e);
+            }
+        }
+        return addresses;
+    }
+
+    @Override
+    public int getRefreshInterval() {
+        if (!this.useLocalConnection) {
+            return 0;
+        }
+        return super.getRefreshInterval();
+    }
+
+    /**
+     * Returns information whether the device-side incoming passwords are to be shown as properties (and if so: in the
+     * clear or replaced with ***)
+     *
+     * @return Configured value
+     */
+    public DeviceSidePasswordDisplayMode getIncludeDeviceSidePasswordsInProperties() {
+        return this.includeDeviceSidePasswordsInProperties;
+    }
+
+    /**
+     * Should the incoming (intercepted) device-side updates be a strict match to local IP (if provided) or hostname
+     * (fallback)
+     *
+     * @return True - if requiring exact match, False - if IP mismatch is allowed
+     */
+    public boolean getMatchAnyIncomingDeviceIp() {
+        return this.matchAnyIncomingDeviceIp;
+    }
+
+    @Override
+    protected String getExtraFieldDescription() {
+        return String.format(
+                "hostname=%s, localDeviceIP=%s, localDevicePort=%d, connectionMode=%s, useLocalConnection=%s, stubServerPort=%d, stubServerListenAddresses=%s, includeDeviceSidePasswordsInProperties=%s, matchAnyIncomingDeviceIp=%s",
+                getOrDefault(this::getHostname), getOrDefault(this::getLocalDeviceIP), localDevicePort, connectionMode,
+                useLocalConnection, stubServerPort, getOrDefault(this::getStubServerListenAddresses),
+                includeDeviceSidePasswordsInProperties, matchAnyIncomingDeviceIp);
+    }
+
+    @Override
+    protected void validateInternal() throws ArgoConfigurationException {
+        if (hostname.isEmpty()) {
+            throw ArgoConfigurationException.forEmptyRequiredParam(ArgoClimaBindingConstants.PARAMETER_HOSTNAME,
+                    i18nProvider);
+        }
+
+        if (!useLocalConnection && connectionMode == ConnectionMode.LOCAL_CONNECTION) {
```

Comment:
Looked at the readme, still not sure. Can't the `useLocalConnection` be dropped at all places? 

`connectionMode == ConnectionMode.LOCAL_CONNECTION` is allready signaling the same isn't it?

Reply:
Nah, it's not. In fact, these 2 parameters (`ConnectionMode.LOCAL_CONNECTION` and `useLocalConnection` are mutually-exclusive).
See:
- the blue (dashed) line on [this diagram](https://github.com/mbronk/openhab-addons/blob/0e011d4b23657565818058dd51dfbf8983e2457e/bundles/org.openhab.binding.argoclima/doc/Argoclima_connection_Advanced_REMOTE_API_PROXY.png)
- Code-side check [here](https://github.com/mbronk/openhab-addons/blob/0e011d4b23657565818058dd51dfbf8983e2457e/bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java#L193-L195)

----
Disclaimer:
I could technically create more modes:
 `LOCAL_CONNECTION`, `REMOTE_API_STUB_WITH_LOCAL_CONNECTION`, **`REMOTE_API_STUB_WITHOUT_LOCAL_CONNECTION`,** `REMOTE_API_PROXY_WITH_LOCAL_CONNECTION`, **`REMOTE_API_PROXY_WITHOUT_LOCAL_CONNECTION`** but found it pretty awkward (and I don't necessairly consider these separate modes - it's more of a flavor of the indirect modes to optionally use local connection as well).

Btw. `useLocalConnection` is an advanced param. Connection mode was as well, but on a 2nd thought, I'll promote it to "regular".